### DISCUSSION TITLE
Node x PHP x .Net, onde e quando usar cada um?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1555] [Perguntas e Respostas]
Fala galera, é mais refletindo e querendo opiniões.

Atualmente estudo NodeJS + ReactJS + React Native.

Meu emprego atual é dev fullstack senior com .Net Core (Senior) + ReactJS (Junior->Pleno) rs
Meu emprego passado, por 2,5 anos foi com PHP (Laravel, Phalcon e Lumen).

Cheguei a conclusão de que:

PHP:
Usar em projetos rápidos, MVPs, sites, alguns ecommerces de pequeno e médio porte. (Uso bastante em projetos autônomos de Blogs, Wordpress ou hotsites)

.Net Core:
Usar em aplicações enterprises, as novidades do .Net Core como multiplataforma, performance e facilidade de implementar APIs me lembram bastante o nodejs, mas ainda prefiro ele para aplicações grande e robustas, os devs são mais antenados em arquitetura, DDD e POO a fundo porque o framework e a linguagem não da muita abertura. 
Usar em projetos enterprises, grandes monolitos, empresas com forte vinculos com microsoft e azure.

NodeJS:
Gosto bastante de fazer projetos de micro-serviços e projetos rápidos, e com certeza projetos em que vão bater muitas requisições por segundo no servidor, é impressionante a produtividade usando node frente a por exemplo usar .net, porém gosto muito de ambas as tecnologias, porém vejo a liberdade dada aqui sendo usada por muitos devs para passar por cima de patterns e padrões, é até temo em um futuro rolar um preconceito igual rola atualmente em certas empresas sobre o PHP.

O que vocês acham, como vocês comparam e em quais ocasiões preferem cada tecnologia?

Vejo empresas que usam somente uma tecnologia pois a equipe só sabe aquela, mas acredito que o cenário ideal é usar a tecnologia que mais se adeque a cada projeto. Vejo o pessoal fazendo sites intitucionais em NodeJS ou em .Net/Razor... e ao me ver estes deveriam ser focados em sistemas, APIs, robos... Sinto uma produtividade muito maior ao criar um site institucional por exemplo com PHP (me parece que ele foi desenvolvido pra isso...).

Gostaria da opnião de vocês.

Abraços!

**Comment by Gustavo-Zamai** [comment]
Caramba , obrigado Wilson, pelo seu depoimento e compartilhar sua experiência, estou entrando agora na área do desenvolvimento, menos de um ano de estudo, e fico muito na duvida entre o que estudar, o que usar, qual vai me trazer um retorno melhor. Atualmente utilizo muito PHP, e vejo muito preconceito da comunidade para com essa linguagem. Novamente agradeço por compartilhar sua experiência.
